src/WebJobs.Script.WebHost/Configuration/ScriptApplicationHostOptionsSetup.cs

```diff
@@ -9,22 +9,23 @@
 
 namespace Microsoft.Azure.WebJobs.Script.WebHost.Configuration
 {
-    public class ScriptApplicationHostOptionsSetup : IConfigureNamedOptions<ScriptApplicationHostOptions>
+    public class ScriptApplicationHostOptionsSetup : IConfigureNamedOptions<ScriptApplicationHostOptions>, IDisposable
     {
         public const string SkipPlaceholder = "SkipPlaceholder";
-
+        private readonly IOptionsMonitorCache<ScriptApplicationHostOptions> _cache;
         private readonly IConfiguration _configuration;
-        private readonly IEnvironment _environment;
-
-        public ScriptApplicationHostOptionsSetup(IConfiguration configuration)
-            : this(configuration, SystemEnvironment.Instance)
-        {
-        }
+        private readonly IOptionsMonitor<StandbyOptions> _standbyOptions;
+        private readonly IDisposable _disposable;
 
-        public ScriptApplicationHostOptionsSetup(IConfiguration configuration, IEnvironment environment)
+        public ScriptApplicationHostOptionsSetup(IConfiguration configuration, IOptionsMonitor<StandbyOptions> standbyOptions,
+            IOptionsMonitorCache<ScriptApplicationHostOptions> cache)
         {
+            _cache = cache ?? throw new ArgumentNullException(nameof(cache));
             _configuration = configuration ?? throw new ArgumentNullException(nameof(configuration));
-            _environment = environment ?? throw new ArgumentNullException(nameof(environment));
+            _standbyOptions = standbyOptions ?? throw new ArgumentNullException(nameof(standbyOptions));
+
+            // If standby options change, invalidate this options cache.
+            _disposable = _standbyOptions.OnChange(o => _cache.Clear());
```

Comment:
With this latest change -- all the IOptionsMonitor invalidation happens on StandbyOptions. Here, we watch for those changes and invalidate the ScriptApplicationHostOptions cache if we see them. This guarantees ordering.